From ca91d85ae92bee325bcd97bce8476d8ec3fd6f3a Mon Sep 17 00:00:00 2001
From: xiongzhu <692949348@qq.com>
Date: Thu, 26 Aug 2021 10:24:23 +0800
Subject: [PATCH] strdup

---
 src/mixer.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/src/mixer.c b/src/mixer.c
index a45502e..0292c51 100644
--- a/src/mixer.c
+++ b/src/mixer.c
@@ -62,6 +62,29 @@
 
 #include "mixer_io.h"
 
+char *strdup(const char *s) {
+    size_t size = strlen(s) + 1;
+    char *p = malloc(size);
+    if (p != NULL) {
+        memcpy(p, s, size);
+    }
+    return p;
+}
+
+char *strndup(const char *s, size_t n) {
+    char *p;
+    size_t n1;
+
+    for (n1 = 0; n1 < n && s[n1] != '\0'; n1++)
+        continue;
+    p = malloc(n + 1);
+    if (p != NULL) {
+        memcpy(p, s, n1);
+        p[n1] = '\0';
+    }
+    return p;
+}
+
 /** A mixer control.
  * @ingroup libtinyalsa-mixer
  */
-- 
2.30.1 (Apple Git-130)

